invenio_records_resources/services/records/service.py

```diff
@@ -464,8 +465,14 @@ def rebuild_index(self, identity, uow=None):
 
         Note: Skips (soft) deleted records.
         """
-        records = self.record_cls.model_cls.query.filter_by(is_deleted=False).all()
-        self.indexer.bulk_index([rec.id for rec in records])
+        model_cls = self.record_cls.model_cls
+        records = (
+            db.session.query(model_cls.id)
+            .filter(model_cls.is_deleted == False)
+            .yield_per(1000)
```

Comment:
does it make sense to pass it as parameter in the method?

Reply:
That's internal SQLAlchemy memory batching/yielding, so in this case it would be 1k UUIDs in memory (so ~16Kb, at any time). This was kind of picked arbitrarily, but any value would be good :)